opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/exponential_histogram/mapping/__init__.py

```diff
@@ -0,0 +1,85 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+
+
+class Mapping(ABC):
+
+    # pylint: disable=no-member
+    def __new__(cls, scale: int):
+
+        with cls._mappings_lock:
+            if scale not in cls._mappings:
+                cls._mappings[scale] = super().__new__(cls)
+
+        return cls._mappings[scale]
+
+    def __init__(self, scale: int) -> None:
+
+        if scale > self._max_scale:
+            raise Exception(f"scale is larger than {self._max_scale}")
+
+        if scale < self._min_scale:
+            raise Exception(f"scale is smaller than {self._min_scale}")
+
+        # The size of the exponential histogram buckets is determined by a
+        # parameter known as scale, larger values of scale will produce smaller
+        # buckets. Bucket boundaries of the exponential histogram are located
+        # at integer powers of the base, where:
+
+        # base = 2 ** (2 ** (-scale))
+        self._scale = scale
+
+    @property
+    @abstractmethod
+    def _min_scale(self) -> int:
+        """
+        Return the smallest possible value for the mapping scale
+        """
+
+    @property
+    @abstractmethod
+    def _max_scale(self) -> int:
+        """
+        Return the largest possible value for the mapping scale
+        """
+
+    @abstractmethod
+    def map_to_index(self, value: float) -> int:
+        """
+        Maps positive floating point values to indexes corresponding to
+        `Mapping.scale`. Implementations are not expected to handle zeros,
+        +inf, NaN, or negative values.
+        """
+
+    @abstractmethod
+    def get_lower_boundary(self, index: int) -> float:
```

Comment:
What's this needed for in the impl of the aggregator? For bucketing in the aggregator, I would have thought you'd only need `map_to_index` and not the other way around.

Reply:
Sorry, what do you mean with "the aggregator"? This is a function of the `Mapping` class.

Reply:
In the future PR where you implement the aggregator using this mapping, hat is `get_lower_boundary()` needed for?

Reply:
Right, we are not using them (for the time being). I see them being used [here](https://github.com/lightstep/otel-launcher-go/blob/c9ca8483be067a39ab306b09060446e7fda65f35/lightstep/sdk/metric/aggregator/histogram/structure/exponential_test.go#L44). Mappings are not part of the public interface so we can remove these functions later if it is better.